pyneuroml/io.py

```diff
@@ -56,9 +59,12 @@ def read_neuroml2_file(
 
     logger.info("Loading NeuroML2 file: %s" % nml2_file_name)
 
-    if not os.path.isfile(nml2_file_name):
+    if not os.path.isfile(nml2_file_name) and critical_includes:
         logger.critical("Unable to find file: %s!" % nml2_file_name)
         sys.exit(FILE_NOT_FOUND_ERR)
+    elif not os.path.isfile(nml2_file_name) and not critical_includes:
```

Comment:
Yes that creates a problem, that's true. I could add another boolean to distinguish between the first iteration and the rest or add 
```python
if already_included is None: 
  already_included = []
  if not os.path.isfile(nml2_file_name):
    logger.critical("Unable to find file: %s!" % nml2_file_name)
    sys.exit(FILE_NOT_FOUND_ERR)
``` 
in the beginning. How does that sound?

Reply:
So, you mean we don't validate using run_jneuroml() but with more specific validations, going 'top to bottom', just before the file is plotted, in order to be able to get more info for the warnings? For example, if plotting a cell, it would only run a morphology validation, but if plotting a network it would run a population validation and a morphology validation as well. Did I get this straight?

Reply:
Yeh, because we know what partial bits of the model we need to visualise it.  For example, we could add things lik e`cell.morphology.validate()` somewhere here:

https://github.com/NeuroML/pyNeuroML/blob/development/pyneuroml/plot/PlotMorphologyVispy.py#L738

Reply:
Alright, I see! But since plot_3D_cell_morphology() is only for plotting detailed cells, maybe we should add them somewhere like here? : 

https://github.com/NeuroML/pyNeuroML/blob/development/pyneuroml/plot/PlotMorphologyVispy.py#L565
and
https://github.com/NeuroML/pyNeuroML/blob/development/pyneuroml/plot/PlotMorphologyVispy.py#L590
and so on

Reply:
Yes---so, a set of tests would be in `plot_3d_cell_morphology`: they'd test if the cell has segments, which are what we plot, and so on. Now that I think of it, I guess we can't directly use `validate()` methods because they don't run such logical tests, so we'll need to add these ourselves: something like: `len(cell.morphology.segments) > 0` (do check the exact variable names).